llvm/test/CodeGen/NVPTX/setmaxnreg.ll

```diff
@@ -0,0 +1,15 @@
+; RUN: llc < %s -march=nvptx64 -mcpu=sm_90a -mattr=+ptx80| FileCheck --check-prefixes=CHECK %s
+; RUN: %if ptxas-12.0 %{ llc < %s -march=nvptx64 -mcpu=sm_90a -mattr=+ptx80| %ptxas-verify -arch=sm_90a %}
+
+declare void @llvm.nvvm.setmaxnreg.sync.aligned.u32(i32 %flags, i32 %reg_count)
+
+; CHECK-LABEL: test_set_maxn_reg
+define void @test_set_maxn_reg() {
+  ; CHECK: setmaxnreg.inc.sync.aligned.u32 96;
+  call void @llvm.nvvm.setmaxnreg.sync.aligned.u32(i32 0, i32 96)
+
+  ; CHECK: setmaxnreg.dec.sync.aligned.u32 64;
+  call void @llvm.nvvm.setmaxnreg.sync.aligned.u32(i32 1, i32 64)
```

Comment:
Do we want to prefer two intrinsic for inc/dec? 
```
llvm.nvvm.setmaxnreg.inc.sync.aligned.u32
llvm.nvvm.setmaxnreg.dec.sync.aligned.u32
```

Reply:
This is what I initially thought, but this would leave us with no option to extend the intrinsic in future (may be with different action settings).

If you think, we should add two explicit intrinsics (since this is just two and not a lot more now), I am fine with that too. Let me know what you think,

Also, I will add verifier logic for the %8 and range checks.

Reply:
I think separate intrinsics make more sense here. The `inc/dec` flags translate into a PTX instruction *name*, so matching intrinsic name would be a better fit, and has a bonus of simplifying the implementation.

Reply:
Sure, updated in the latest revision,